### ISSUE TITLE
Fails to build Windows and Mac releases

### ISSUE THREAD

**Issue by hollunder** [MisterDA/love-release#67]
Windows:
`$ ~/.luarocks/bin/love-release -W 64 -a murks -e murks@freeshell.de --uti "at.murks.wumpus" -t "Blind Wumpus World" -u "https://murks.itch.io/blind-wumpus-world" -d "A short Wumpus World game made during trijam #27" -p "blind-wumpus-world" -v 3 builds code
{
  title = 'Blind Wumpus World',
  package = 'blind-wumpus-world',
  loveVersion = '11.0',
  version = '3',
  author = 'murks',
  email = 'murks@freeshell.de',
  description = 'A short Wumpus World game made during trijam #27',
  homepage = 'https://murks.itch.io/blind-wumpus-world',
  identifier = 'at.murks.wumpus',
  excludeFileList = { } ,
  compile = false,
  projectDirectory = '/home/murks/trijam/20190713/code',
  releaseDirectory = '/home/murks/trijam/20190713/builds',
}
/usr/bin/lua5.3: ....luarocks/share/lua/5.3/love-release/scripts/windows.lua:40: HTTP/1.1 404 Not Found
stack traceback:
	[C]: in function 'assert'
	....luarocks/share/lua/5.3/love-release/scripts/windows.lua:40: in upvalue 'release'
	....luarocks/share/lua/5.3/love-release/scripts/windows.lua:85: in function 'love-release.scripts.windows.script'
	(...tail calls...)
	...urks/.luarocks/share/lua/5.3/love-release/pipes/args.lua:109: in local 'args'
	...uarocks/rocks-5.3/love-release/2.0.12-2/bin/love-release:11: in main chunk
	[C]: in ?
`

Macintosh:
`$ ~/.luarocks/bin/love-release -M -a murks -e murks@freeshell.de --uti "at.murks.wumpus" -t "Blind Wumpus World" -u "https://murks.itch.io/blind-wumpus-world" -d "A short Wumpus World game made during trijam #27" -p "blind-wumpus-world" -v 3 builds code 
{
  title = 'Blind Wumpus World',
  package = 'blind-wumpus-world',
  loveVersion = '11.0',
  version = '3',
  author = 'murks',
  email = 'murks@freeshell.de',
  description = 'A short Wumpus World game made during trijam #27',
  homepage = 'https://murks.itch.io/blind-wumpus-world',
  identifier = 'at.murks.wumpus',
  excludeFileList = { } ,
  compile = false,
  projectDirectory = '/home/murks/trijam/20190713/code',
  releaseDirectory = '/home/murks/trijam/20190713/builds',
}
/usr/bin/lua5.3: /home/murks/.luarocks/share/lua/5.3/love-release/utils.lua:58: attempt to compare nil with number
stack traceback:
	/home/murks/.luarocks/share/lua/5.3/love-release/utils.lua:58: in metamethod '__le'
	.../.luarocks/share/lua/5.3/love-release/scripts/macosx.lua:31: in function 'love-release.scripts.macosx.script'
	(...tail calls...)
	...urks/.luarocks/share/lua/5.3/love-release/pipes/args.lua:98: in local 'args'
	...uarocks/rocks-5.3/love-release/2.0.12-2/bin/love-release:11: in main chunk
	[C]: in ?`

Besides the documentation, doesn't mention a couple of things, that the description is required for DEB, --uti for MAC and how to actually use the conf.lua.

**Comment by MisterDA** [comment]
Hello!
Thanks for the bug report. I have pushed fixes and more debugging output. Could you please test with latest master?

```sh
luarocks install --dev love-release # --local
```

4682ffe should fix the issue with the windows download.
I think I know where the issue with macOS comes from, but I wasn’t able to reproduce it.
I’ve added notes about required parameters in ff33b0f.

About the configuration, it is explained in the README: just add a subtable `releases` to the `t` table that is passed to `love.conf`. The fields are listed.

> These informations can be stored in your `conf.lua` file to be used later.
>
> ```lua
> function love.conf(t)
>   t.releases = { --[[ … ]]-- }
> ```

**Comment by hollunder** [comment]
Just to be clear, installing through luarocks should get me that?

```
$ luarocks install love-release --local           
Installing https://luarocks.org/love-release-2.0.12-2.rockspec

Cloning into 'love-release'...
remote: Enumerating objects: 26, done.
remote: Counting objects: 100% (26/26), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 26 (delta 2), reused 7 (delta 0), pack-reused 0
Receiving objects: 100% (26/26), 16.67 KiB | 16.67 MiB/s, done.
Resolving deltas: 100% (2/2), done.

Error: Build error: Failed installing src/scripts/macosx.lua in /home/murks/.luarocks/lib/luarocks/rocks-5.3/love-release/2.0.12-2/lua/love-release/scripts/macosx.lua: The source and destination are the same files
```

**Comment by MisterDA** [comment]
No. I have the same bug with 2.0.12-2 now, I will investigate. But could you try the `--dev` flag? It will install love-release straight from master with the latest commits.

**Comment by hollunder** [comment]
Thanks!
The Windows issue seems to be gone (note that I can't test whether they actually work, but something got built without error).
